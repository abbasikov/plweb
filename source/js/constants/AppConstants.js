/*
 * AppConstants
 * These are the variables that determine what our central data store (reducer.js)
 * changes in our state. When you add a new action, you have to add a new constant here
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'YOUR_ACTION_CONSTANT';
 */

// The initial application state
export const INITIAL_STATE = {
    formState: {
        username: '',
        password: ''
    },
    sendingLoginRequest: false,
    loggedIn: false,
    errorMessage: '',
    currentUser:{},
    loginError:'',
    homeScreenLoading:false,
    feeds:[]
};

export const SERVER_URL = 'https://demo5100262.mockable.io';
export const CHANGE_FORM = 'CHANGE_FORM';
export const SET_AUTH = 'SET_AUTH';
export const SET_FEEDS = 'SET_FEEDS';
export const SENDING_REQUEST = 'SENDING_REQUEST';
export const SENDING_LOGIN_REQUEST = 'SENDING_LOGIN_REQUEST';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
export const FETCH_POSTS = 'FETCH_POSTS';
export const CREATE_POST = 'CREATE_POST';
export const FETCH_POST = 'FETCH_POST';
export const DELETE_POST = 'DELETE_POST';
export const SELECT_BOOK = "SELECT_BOOK";
export const LOGIN_USER = "LOGIN_USER";
export const SET_HOME_SCREEN_LOADING = "SET_HOME_SCREEN_LOADING";