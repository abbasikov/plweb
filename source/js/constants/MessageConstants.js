/*
 * MessageConstants
 * These are the variables that contain the text which is displayed on certain errors
 *
 * Follow this format:
 * export const YOUR_CONSTANT = 'Your message text';
 */
export const FIELD_MISSING = 'Please fill out the entire form.';
export const WRONG_PASSWORD = 'Wrong password.';
export const USER_NOT_FOUND = 'This username does not exist.';
export const USERNAME_TAKEN = 'Sorry, but this username is already taken';
export const GENERAL_ERROR = 'Something went wrong, please try again';
export const LOADING_TEXT= "Please wait...";