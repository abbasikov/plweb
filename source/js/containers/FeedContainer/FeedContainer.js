import React from 'react';
import Feed from '../../components/Feed/Feed';
import { browserHistory } from 'react-router';
import routes from '../../routes';
import { connect } from 'react-redux';
import FeedLoader from '../../components/FeedLoader/FeedLoader';

class FeedContainer extends React.Component {

    static propTypes = {
        feeds:React.PropTypes.array
    };

    constructor(props) {
        super(props);
    }

    getFeeds() {
        var allFeeds = [];
        if(this.props.feeds) {
            allFeeds = this.props.feeds.map(function(feed) {
                return <Feed key={feed.userName}
                             userName={feed.userName}
                             content={feed.content}
                             boardName={feed.boardName} />;
            });
        }

        return allFeeds;
    }

    render() {
        return (
            <div className="content-wrapper">
                <div className="panel panel-flat ">
                    <div className="panel-body panel-post">
                        <button type="button" className="btn bg-orange-600 btn-xs"
                                onClick={(e) => browserHistory.push(routes.home.childRoutes.createPost.path)}>
                            <i className="glyphicon glyphicon-plus position-left"></i>
                            <span>POST</span>
                        </button>
                    </div>
                </div>

                {this.props.feeds.length == 0 ? <FeedLoader /> : this.getFeeds() }


            </div>
        );
    }
}

function mapStateToProps(state) {
    //Whatever is returned from this function will end up
    // as props inside of this container component
    return {
        feeds:state.feeds
    }
}

//function mapDispatchToProps(dispatch) {
//    return bindActionCreators({logoutUser,setHomeScreenLoading}, dispatch);
//}

export default connect(mapStateToProps, null)(FeedContainer);