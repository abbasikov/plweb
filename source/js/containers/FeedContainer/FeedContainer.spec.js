import React from 'react';
import {shallow, render, mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import FeedContainer from './FeedContainer';
import toJson  from 'enzyme-to-json';
import { INITIAL_STATE } from '../../constants/AppConstants';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([ thunk ]);

describe('FeedContainer', function() {
    var store;
    var container;

    beforeEach(function() {
        store = mockStore(INITIAL_STATE);
    });

    describe('#renders', () => {

        it('when feeds are empty', function() {
            container = render(<Provider store={store}><FeedContainer /></Provider>);
            expect(toJson(container)).toMatchSnapshot();
        });

        it('when feeds are not empty', function() {
            var mockedFeeds = [{
                userName:'test',
                content:'test',
                boardName:'test'
            }];
            store.getState().feeds = mockedFeeds;
            container = render(<Provider store={store}><FeedContainer /></Provider>);
            expect(toJson(container)).toMatchSnapshot();
        });
    });
});