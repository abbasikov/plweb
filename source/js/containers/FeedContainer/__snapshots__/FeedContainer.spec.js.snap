// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedContainer #renders when feeds are empty 1`] = `
<div
  class="content-wrapper"
>
  <div
    class="panel panel-flat "
  >
    <div
      class="panel-body panel-post"
    >
      <button
        class="btn bg-orange-600 btn-xs"
        type="button"
      >
        <i
          class="glyphicon glyphicon-plus position-left"
        />
        <span>
          POST
        </span>
      </button>
    </div>
  </div>
  <div
    class="l1-core-rail"
  >
    <div
      class="panel post-feed delay260"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
    <div
      class="panel post-feed delay320"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
    <div
      class="panel post-feed delay380"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
  </div>
</div>
`;

exports[`FeedContainer #renders when feeds are not empty 1`] = `
<div
  class="content-wrapper"
>
  <div
    class="panel panel-flat "
  >
    <div
      class="panel-body panel-post"
    >
      <button
        class="btn bg-orange-600 btn-xs"
        type="button"
      >
        <i
          class="glyphicon glyphicon-plus position-left"
        />
        <span>
          POST
        </span>
      </button>
    </div>
  </div>
  <div
    class="panel panel-flat border-top-warning"
  >
    <div
      class="panel-heading"
    >
      <h5
        class="panel-title feed-owner"
      >
        test
        <span
          class="label border-warning-500 text-warning-500"
        >
          Author
        </span>
      </h5>
      <div
        class="heading-elements selected-course"
      >
        <span
          class="label border-left-warning-500 label-striped color-warning-500"
        >
          test
        </span>
      </div>
    </div>
    <div
      class="panel-body"
    >
      <h6
        class="text-semibold"
      >
        Start your development with no hassle!
      </h6>
      <p
        class="content-group"
      >
        test
      </p>
    </div>
  </div>
</div>
`;
